packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/AppTests.kt

```diff
@@ -292,42 +301,122 @@ class AppTests {
 //        assertEquals(registry, app.getFunctions(user, registry).defaultCodecRegistry)
 //    }
 //
-//    @Test
-//    fun encryption() {
-//        val context = InstrumentationRegistry.getInstrumentation().targetContext
-//
-//        // Create new test app with a random encryption key
-//        val testApp = TestApp(appName = TEST_APP_2, builder = {
-//            it.encryptionKey(TestHelper.getRandomKey())
-//        })
-//
-//        try {
-//            // Create Realm in order to create the sync metadata Realm
-//            var user = testApp.login(Credentials.anonymous())
-//
-//            val syncConfig = SyncConfiguration
-//                .Builder(user, "foo")
-//                .testSchema(SyncStringOnly::class.java)
-//                .build()
-//
-//            Realm.getInstance(syncConfig).close()
-//
-//            // Create a configuration pointing to the metadata Realm for that app
-//            val metadataDir = File(context.filesDir, "mongodb-realm/${testApp.configuration.appId}/server-utility/metadata/")
-//            val config = RealmConfiguration.Builder()
-//                .name("sync_metadata.realm")
-//                .directory(metadataDir)
-//                .build()
-//            assertTrue(File(config.path).exists())
-//
-//            // Open the metadata realm file without a valid encryption key
-//            assertFailsWith<RealmFileException> {
-//                DynamicRealm.getInstance(config)
-//            }
-//        } finally {
-//            testApp.close()
-//        }
-//    }
+
+    @Test
+    fun encryptedMetadataRealm() {
+        // Create new test app with a random encryption key
+        val key = TestHelper.getRandomKey()
+        val app = TestApp(
+            appName = TEST_APP_FLEX,
+            builder = {
+                it
+                    .encryptionKey(key)
+                    .syncRootDirectory("${appFilesDirectory()}/foo")
+            }
+        )
+
+        try {
+            // Create Realm in order to create the sync metadata Realm
+            val user = app.createUserAndLogin()
+            val syncConfig = SyncConfiguration
+                .Builder(user, setOf(ParentPk::class, ChildPk::class))
+                .build()
+            Realm.open(syncConfig).close()
+
+            // Create a configuration pointing to the metadata Realm for that app
+            val metadataDir = "${app.configuration.syncRootDirectory}/mongodb-realm/${app.configuration.appId}/server-utility/metadata/"
+            val config = RealmConfiguration
+                .Builder(setOf())
+                .name("sync_metadata.realm")
+                .directory(metadataDir)
+                .encryptionKey(key)
+                .deleteRealmIfMigrationNeeded()
```

Comment:
Without this there will be an error `Caused by: io.realm.kotlin.internal.interop.RealmCoreLogicException: [18]: Provided schema version 0 is less than last set version 6.`

Reply:
Hmm, that line actually invalidates the test, because the Realm gets deleted now.

Does it work if you set `RealmConfiguration.schemaVersion(6)` instead?

Reply:
But doesn't it recreate the realm with the new schema? I'm also checking `assertTrue(fileExists(config.path))` which is the path to the `sync_metadata.realm` file.

Yes schemaVersion(6) works, but would be nice to not hard code 6 if that was to change?